### ISSUE TITLE
Move to ITRB

### ISSUE THREAD

**Issue by cbizon** [ExposuresProvider/cam-kp-api#550]
From Architecture call today:
https://arax.ncats.io/?smartapi=1 shows that CAM-KP is not yet in ITRB Prod.  Please deploy to prod ASAP.

Also, before long, we will also want staging and testing instances on ITRB.

**Comment by gaurav** [comment]
We do have (older versions of) CAM-KP-API on ITRB prod, staging and testing:
- https://cam-kp-api.transltr.io/1.2.0/docs/index.html?url=docs.yaml (prod)
- https://cam-kp-api.test.transltr.io/1.2.0/docs/index.html?docs=docs.yaml (test, currently down)
- https://cam-kp-api.ci.transltr.io/1.2.0/docs/index.html?url=docs.yaml (dev)

(All of these currently talk to the SPARQL endpoint hosted at RENCI; we're thinking about ways of moving the Blazegraph server to ITRB as well.)

I notice that these instances aren't listed at https://smart-api.info/ either -- if I submit them there, will that propagate back to https://arax.ncats.io or is there another list we should update there?

**Comment by cbizon** [comment]
Yes, if they are registered in smart-api registry then that will take care of everything.

**Comment by gaurav** [comment]
I've now added [`cam-kp-api.transltr.io`](https://smart-api.info/api/metadata/4fb4446816d08d2b84e08462c051c8f8) and [`cam-kp-api.ci.transltr.io`](https://smart-api.info/ui/80558f3abf621ffbd6d4cf7b96237180) to SmartAPI, and confirmed that they now show up in https://arax.ncats.io/?smartapi=1. Unfortunately, because of a bug in CAM-KP-API, they all report themselves as `production` instances of `cam-kp-api.renci.org`, but you can see they are working correctly by looking at the raw metadata.

I'm currently working with Yaphet to deploy the latest CAM-KP-API -- which should fix this bug -- to ITRB Develop via https://github.com/helxplatform/translator-devops/pull/450. We tried this today but ran into an error that I hope I've now fixed. I'll keep iterating on this and report back on this issue once we've updated it correctly. Once that's done, we'll try to push this update to ITRB Test and ITRB Prod.

CAM-KP-API instances currently access the Blazegraph instance hosted on Sterling in RENCI to make SPARQL queries. @balhoff and I have been discussing how to move that to ITRB as well so that CAM-KP-API can be fully hosted on ITRB. Jim is currently working on moving Ontology-KP to ITRB, and seems to be making good progress; once he's done that, I'll use his code to set up a similar system for CAM-KP-API.

https://cam-kp-api.test.transltr.io/1.2.0/docs/index.html?docs=docs.yaml is still down, but I'm hoping that once we get PR https://github.com/helxplatform/translator-devops/pull/450 deployed correctly, we can redeploy it to all ITRB instances.

So, currently outstanding:
- [ ] Deploy CAM-KP-API 0.1.2-trapi-1.2 to ITRB (1-2 days)
- [ ] Ensure that CAM-KP-API instances on RENCI and ITRB are reporting the correct version and maturity level (1 day)
- [ ] Deploy backend Blazegraph instance to ITRB (3-4 days)

**Comment by gaurav** [comment]
CAM-KP-API 0.1.2 is now on ITRB-CI: https://smart-api.info/ui/80558f3abf621ffbd6d4cf7b96237180
I'm not working with Yaphet on getting that version to ITRB-test and ITRB-prod as well.

**Comment by gaurav** [comment]
CAM-KP-API v0.1.2 is now on ITRB-Prod too: https://smart-api.info/ui/4fb4446816d08d2b84e08462c051c8f8
I'm not sure why ITRB-test isn't working; debugging that now.

**Comment by gaurav** [comment]
CAM-KP-API v0.1.2 is now on ITRB-Test too: https://smart-api.info/registry?q=dbc3bd68479f4a2bbf641ec45c7c013c
